numpydoc/tests/test_docscrape.py

```diff
@@ -1199,6 +1199,33 @@ def test_templated_sections():
     """)
 
 
+def test_nonstandard_property():
+    # test discovery of a property that does not satisfy isinstace(.., property)
+
+    class SpecialProperty(object):
+
+        def __init__(self, axis=0, doc=""):
+            self.axis = axis
+            self.__doc__ = doc
+
+        def __get__(self, obj, type):
+            if obj is None:
+                # Only instances have actual _data, not classes
+                return self
+            else:
+                return obj._data.axes[self.axis]
```

Comment:
I added a test case. 
The line above is from our original example in pandas, and actually does not matter (it's the `if obj is None: return self` that matters, so I can also remove this